### PR TITLE
unique service names (main)

### PR DESCRIPTION
**Pull request by wind57**

### CODE REVIEW

File: spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClient.java [hunk 1]
```diff
@@ -162,13 +162,13 @@ private List<ServiceInstance> getNamespaceServiceInstances(EndpointSubsetNS es,
 
 	@Override
 	public List<String> getServices() {
```
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClient.java:164]: if we have for example enabled `all-namespaces=true` and there is such instances:

- `my-service` in namespace `a`
- `my-service` in namespace `b`

a call to `getServices` will currently return `[my-service, my-service]`, which imo, does not make sense to the callers. wdyt?

**Comment by wind57** [comment]
@ryanjbaxter I guess a minor thing to look at. thank you.

**Comment by ryanjbaxter** [comment]
Wait I thought I said they should be unique?

**Comment by wind57** [comment]
> Yes we should be returning duplicates of the same name, because they would be considered the same service

did I miss-understood your comment?

**Comment by ryanjbaxter** [comment]
Oops yes I have a typo there..."should NOT be returning duplicates"

**Comment by ryanjbaxter** [comment]
We should fix this in 2.1.x

**Comment by wind57** [comment]
no worries, I'll create a PR for that also.

**Comment by ryanjbaxter** [comment]
We should just need one and merge it forward

**Comment by wind57** [comment]
but the branches have diverged a lot in this area. wouldn't it be easier to have two PRs?

**Comment by ryanjbaxter** [comment]
2 PRs is fine, i still have to merge 2.1.x into main though, but its no big deal to me

**Comment by wind57** [comment]
in such a case, can we merge this one? :)

**Comment by ryanjbaxter** [comment]
I want to merge the 2.1.x version first because like I said I need to merge that forward (unnecessarily).  Are there tests that need to be changed in that branch?

**Comment by wind57** [comment]
no, iirc, there are not test changes.

But then, I'd like to make sure that the tests indeed pass on `2.1.x`. So how about I create a PR to have a pipeline in `2.1.x`, first? I would hate to push something without the entire pipeline to run. I'll make this my priority.

**Pull request merged**